ui/src/features/Editor/components/Canvas/useNodes.ts

```
@@ -14,67 +13,70 @@ import { MouseEvent, useCallback } from "react";
               source,
               target,
-            })),
-          );
+            },
+          })),
+        );
 
-          return [...remainingEdges, ...createdEdges];
-        }, edges),
-      );
+        return [...acc, ...removals, ...additions];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)